open_parser/base.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+import requests
+
+CAMBIO_UPLOAD_URL = (
+    "https://p1iz3c1c77.execute-api.us-west-2.amazonaws.com/v1/cambio_api/upload"
+)
+CAMBIO_EXTRACT_URL = (
+    "https://p1iz3c1c77.execute-api.us-west-2.amazonaws.com/v1/cambio_api/extract"
+)
+
+
+class OpenParser:
+    def __init__(self, apiKey="") -> None:
+        self._uploadurl = CAMBIO_UPLOAD_URL
+        self._extracturl = CAMBIO_EXTRACT_URL
+        self._request_header = {"x-api-key": apiKey}
+
+    def setAPIKey(self, apiKey):
+        self._request_header = {"x-api-key": apiKey}
+
+    def extract_pdf_content(self, file_path):
+        user_id, job_id, s3_key = self._request_and_upload_by_apiKey(file_path)
```

Comment:
qq: consider upload have async extract behind the scene, can the next `self_request_file_extraction` file due to the file is just not ready? If so, what is the best practice here?